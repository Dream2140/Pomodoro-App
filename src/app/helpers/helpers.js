/**
 * @description Helper Functions
 * @exports Helpers
 * @class Helpers
 */
export class Helpers {

  /**
   * @description sorts object in special key order
   * for rendering buttons template in settings component
   * @param {object} obj
   * @return {object} ordered object
   * @memberof Helpers
   */
  static sortObjectByKeys(obj) {
    const ordered = {};
    const keysOrder = ['work', 'iteration', 'shortBreak', 'longBreak'];

    keysOrder.forEach(item => {
      Object.keys(obj).forEach(key => {
        if (key === item) {
          ordered[key] = obj[key];
        }
      });
    });

    return ordered;
  }

  /**
   * @description checks if it is a new user
   * @return {undefined}
   * @memberof Helpers
   */
  static firstSession() {
    const flag = JSON.parse(sessionStorage.getItem('isNewUser'));
    
    if (flag === null) {
      sessionStorage.setItem('isNewUser', true);
    } else if (!flag) {
      return;
    } else {
      sessionStorage.setItem('isNewUser', false);
    }
  }

  /**
   * @description creates object for tasks
   * @return {object} empty with task categories
   * @memberof Helpers
   */
  static createCategoriesObj() {
    const taskCategoriesProto = {
      work: {},
      education: {},
      hobby: {},
      sport: {},
      other: {}
    };

    return taskCategoriesProto;
  }

  /**
   * @description generate unique id for task
   * @return {string} unique id 
   * @memberof Helpers
   */
  static generateUniqueId() {
    return Date.now().toString(36) + Math.random().toString(36).substr(2);
  }

  /**
   * @description transforms date into format 'MM-dd-YY'
   * @param {object} date current date
   * @return {string} date in format 'MM-dd-YY'
   * @memberof Helpers
   */
  static getCurrentDate(date) {
    let currentDate = date ? new Date(date) : new Date();
    
    return `${currentDate.getMonth() < 9
      ? '0' + (currentDate.getMonth() + 1)
      : currentDate.getMonth() + 1
      }-${currentDate.getDate() < 10
        ? '0' + currentDate.getDate()
        : currentDate.getDate()
      }-${currentDate
        .getFullYear()
        .toString()
        .slice(-2)}`;
  }

  /**
   * @description returns list of month days
   * @return {object}
   * @memberof Helpers
   */
  static getNumberDate(date) {

    const dateLength = 8,
      startIndexOfYear = 6,
      startIndexOfMonth = 0,
      endIndexOfMonth = 3,
      startIndexOfDay = 3,
      endIndexOfDay = 5,
      TWO_THOUSANDS_YEARS = 2000;

    if (!date || date.length !== dateLength) {
      return;
    }

    const year = parseInt(date.substring(startIndexOfYear), 10) + TWO_THOUSANDS_YEARS;
    const month = parseInt(date.substring(startIndexOfMonth, endIndexOfMonth), 10) -1 ;
    const day = parseInt(date.substring(startIndexOfDay, endIndexOfDay), 10);
    const result = new Date(year, month, day);
    
    return result;

  }

  static getMonthDays() {
    const DAYS_IN_MONTH = 30;
    const monthDaysList = [];

    for (let i = 1; i <= DAYS_IN_MONTH; i++) {
      monthDaysList.push(i);
    }

    return monthDaysList;
  }

  /**
  * @description returns list of weekdays
  * @return {object}
  * @memberof Helpers
  */
  static getWeekDays() {
    const DAYS_IN_WEEK = 7;
    const MILLISECONDS_IN_DAY = 86400000;
    const weekDaysList = [];
    const currentDate = new Date();
    let currentTime = currentDate.getTime();
    const options = {
      weekday: 'short'
    };

    for (let i = 0; i < DAYS_IN_WEEK; i++) {
      weekDaysList.push(
        Intl.DateTimeFormat('en-US', options).format(currentDate)
      );
      currentTime -= MILLISECONDS_IN_DAY;
      currentDate.setTime(currentTime);
    }

    return weekDaysList.reverse();
  }

  /**
   * @description returns current week day by date
   * @param {object} date
   * @return {string} week day
   * @memberof Helpers
   */
  static getCurrentWeekDayByDate(date) {
    const currentDate = Helpers.getNumberDate(date);
    if (!currentDate) {
      return;
    }
    const options = {
      weekday: 'short'
    };

    return Intl.DateTimeFormat('en-US', options).format(currentDate);
  }

  /**
   * @description returns day index by date
   * @param {string} date
   * @return {number} index from 30 to 1
   * @memberof Helpers
   */
  static getCurrentMonthDayByDate(date) {
    const currentDayDate = Helpers.getNumberDate(date);
    if (!currentDayDate) {
      return;
    }
    const currentDate = new Date();
    const MILLISECONDS_IN_DAY = 86400000;
    const DAYS_IN_MONTH = 30;
    let currentDayIndex = 30;

    const difference = currentDate.getTime() - currentDayDate.getTime();

    if (difference > MILLISECONDS_IN_DAY) {
      currentDayIndex =
        DAYS_IN_MONTH - Math.floor(difference / MILLISECONDS_IN_DAY);
    }

    return currentDayIndex;
  }


}
