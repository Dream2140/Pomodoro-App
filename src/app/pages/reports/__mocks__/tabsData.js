export const tabsData = {
    leftTabs: false,
    typeTabs: 'reports__top-tabs',
    tabPosition: 'tab__item--centered',
    tabData: [

      {
        period: true,
        val: 'day',
        data: 'Day',
        dataTab: 'data-reports-period="day"'
      },
      {
        period: true,
        val: 'week',
        data: 'Week',
        dataTab: 'data-reports-period="week"'
      },
      {
        period: true,
        val: 'month',
        data: 'Month',
        dataTab: 'data-reports-period="month"'
      }
    ]
};
