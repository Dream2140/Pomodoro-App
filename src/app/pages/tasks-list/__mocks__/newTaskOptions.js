export const newTaskOptions = {
  categoryId: 'other',
  completeDate: null,
  completedCount: [],

  deadline: 'April 5, 2022',
  
  description: 'Hello! Let`s start cleaning!',
  estimation: '5',
  failedPomodoros: [],
  priority: 'middle',
  status: 'GLOBAL_LIST',
  title: 'Test',
};
