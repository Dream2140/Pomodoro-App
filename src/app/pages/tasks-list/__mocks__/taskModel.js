export class TaskModel {
  constructor(options) {
    this.options = {
      title : options.title,
      description : options.description,
      failedPomodoros : options.failedPomodoros,
      completeDate : options.completeDate,
      completedCount : options.completedCount,
      priority : options.priority,
      categoryId : options.categoryId,
      status : options.status,
      estimation : options.estimation,
      deadline : options.deadline,
    }
  }
}