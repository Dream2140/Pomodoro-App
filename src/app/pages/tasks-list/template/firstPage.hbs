{{> header }}
    <main class="content task-list-page">
        <section class="first-message message">
            <div class="wrapper">
                <div class="message__container">
                    <img src="../../images/tomato_settings.svg" alt="Logo" class="message__img">
                    <div class="message__text message__text--bottom-margin">
                        <p>As you visited site for a first time you can
                            check and customize your default
                            application settings
                        </p>
                    </div>
                    <div class="message__buttons service-btn">
                        <button class="service-btn__btn service-btn__btn--blue message__buttons-skip">Skip</button>
                        <button class="service-btn__btn service-btn__btn--green message__buttons-settings">Go to settings</button>
                    </div>
                </div>
            </div>
        </section>


    </main>
