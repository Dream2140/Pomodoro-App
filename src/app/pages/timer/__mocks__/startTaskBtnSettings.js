export const startTaskBtnSettings = {
    secondBtn: true,
    firstColor: 'red',
    firstServiceClass: 'fail-pomodora',
    firstData: 'Fail Pomodora',
    secondColor: 'green',
    secondServiceClass: 'finish-pomodora',
    secondData: 'Finish Pomodora'
};