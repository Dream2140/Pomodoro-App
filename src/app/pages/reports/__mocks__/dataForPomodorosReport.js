import { Helpers } from '../../../helpers/helpers';

export const dataForPomodorosReport = [
  {
    date: Helpers.getCurrentDate(),
    failed: 7,
    high: 1,
    low: 0,
    middle: 0,
    urgent: 3,
  },
];
