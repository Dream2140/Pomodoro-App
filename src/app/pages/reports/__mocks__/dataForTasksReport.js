import { Helpers } from '../../../helpers/helpers';

export const dataForTasksReport = [
  {
    date: Helpers.getCurrentDate(),
    failed: 1,
    high: 0,
    low: 0,
    middle: 0,
    urgent: 1,
  },
];
