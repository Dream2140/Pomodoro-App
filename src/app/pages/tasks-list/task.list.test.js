/* eslint-disable no-undef */
import { TasksListModel } from './scripts/tasksListModel';
import { settingsData } from '../../helpers/settingsData';
import allTasksDone from './template/all-tasks-done.hbs';
import emptyPage from './template/add-firts-task.hbs';
import firstPage from './template/firstPage.hbs';
import noDailyTasks from './template/task-added.hbs';
import noTasksLeft from './template/no-tasks.hbs';
import taskListContainer from './template/tasksList.hbs';
import { tasks } from './__mocks__/tasks';
import { tasksFiltered } from './__mocks__/tasksFiltered';
import { categories } from './__mocks__/categories';
import { objProto } from './__mocks__/objProto';
import { newTaskOptions } from './__mocks__/newTaskOptions';
import { TaskModel } from './__mocks__/taskModel';

const tasksListModel = new TasksListModel(settingsData);
tasksListModel.taskListCollection = tasks;

describe('rendering', () => {
    it('renders correctly allTasksDone', () => {
        const tree = JSON.stringify(allTasksDone());
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly emptyPage', () => {
        const tree = JSON.stringify(emptyPage());
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly firstPage', () => {
        const tree = JSON.stringify(firstPage(tasksListModel.taskListTemplateData));
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly noDailyTasks', () => {
        const tree = JSON.stringify(noDailyTasks());
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly noTasksLeft', () => {
        const tree = JSON.stringify(noTasksLeft());
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly taskListContainer', () => {
        const tree = JSON.stringify(
            taskListContainer(tasksListModel.taskListTemplateData)
        );
        expect(tree).toMatchSnapshot();
    });
});

describe('createTaskListCollection', () => {
    it('should create task-list', () => {
        tasksListModel.createTaskListCollection();
        expect(tasksListModel.tasksCollection).toEqual(tasksFiltered);
    });
});

describe('createCategories', () => {
    it('should create categories', () => {
        const res = tasksListModel.createCategories([tasks[0]], objProto);
        expect(res).toEqual(categories);
    });

    it('should return falsy result', () => {
        const res = tasksListModel.createCategories([tasks[0]]);
        expect(res).toBeFalsy();
    });

    it('should return falsy result', () => {
        const res = tasksListModel.createCategories();
        expect(res).toBeFalsy();
    });

    it('should return falsy result', () => {
        const res = tasksListModel.createCategories('test', true);
        expect(res).toBeFalsy();
    });
});

describe('createNewTask', () => {
    it('should create new task', () => {
        jest.spyOn(tasksListModel, 'createTaskListCollection');
        jest.spyOn(tasksListModel, 'sendTaskData');

        const res = new TaskModel(newTaskOptions);
        tasksListModel.createNewTask(newTaskOptions);

        expect(
            tasksListModel.taskListCollection[
            tasksListModel.taskListCollection.length - 1
            ]
        ).toMatchObject(res.options);
        expect(tasksListModel.createTaskListCollection).toBeCalled();
        expect(tasksListModel.sendTaskData).toBeCalled();
    });

    it('should return falsy result', () => {
        const res = tasksListModel.createNewTask();
        expect(res).toBeFalsy();
    });

    it('should return falsy result', () => {
        const res = tasksListModel.createNewTask(2343434);
        expect(res).toBeFalsy();
    });

    it('should return falsy result', () => {
        const res = tasksListModel.createNewTask(true);
        expect(res).toBeFalsy();
    });
});

describe('getCurrentDate', () => {
    it('should return current day obj', () => {
        const res = tasksListModel.getCurrentDate();
        expect(res).toBeTruthy();
    });

    it('should return day obj', () => {
        const today = new Date(2022, 2, 26);
        const options = {
            month: 'long',
        };
        const newDate = {
            fullDeadline: today.getTime(),
            month: Intl.DateTimeFormat('en-US', options).format(today),
            day: today.getDate(),
        };

        const res = tasksListModel.getCurrentDate('March 26, 2022');
        expect(newDate).toEqual(res);
    });

    it('should return current day obj', () => {
        const res = tasksListModel.getCurrentDate(true);
        expect(res).toBeTruthy();
    });

    it('should return current day obj', () => {
        const res = tasksListModel.getCurrentDate('');
        expect(res).toBeTruthy();
    });
});

describe('updateTask', () => {
    it('should update current task', () => {
        jest.spyOn(tasksListModel, 'createTaskListCollection');
        jest.spyOn(tasksListModel, 'sendTaskData');

        const newTask = tasks[1];
        const newDescription = 'Wow! This is the greatest test ever!!!';
        newTask.description = newDescription;
        tasksListModel.updateTask(newTask);
        const updatedTask = tasksListModel.taskListCollection.find(
            item => item.id === newTask.id
        );
        expect(updatedTask.description).toBe(newDescription);
        expect(tasksListModel.createTaskListCollection).toBeCalled();
        expect(tasksListModel.sendTaskData).toHaveBeenCalledWith(
            'tasks',
            newTask,
            newTask.id
        );
    });

    it('should return falsy value', () => {
        const res = tasksListModel.updateTask();
        expect(res).toBeFalsy();
    });
});

describe('changeTaskStatus', () => {
    it('should change task status', () => {
        jest.spyOn(tasksListModel.settingsData, 'sendData');

        const id = tasks[0].id;
        const newStatus = 'TEST';
        tasksListModel.changeTaskStatus(id, newStatus);
        const updatedTask = tasksListModel.taskListCollection.find(
            item => item.id === id
        );
        expect(updatedTask.status).toBe(newStatus);
        expect(tasksListModel.settingsData.sendData).toHaveBeenCalledWith(
            'tasks',
            updatedTask,
            id
        );
    });

    it('should return falsy value', () => {
        const res = tasksListModel.changeTaskStatus();
        expect(res).toBeFalsy();
    });
});

describe('changeTaskStatus', () => {
    it('should change task status', () => {
        jest.spyOn(tasksListModel.settingsData, 'removeItem');
        jest.spyOn(tasksListModel.renderNoTasksLeftPageEvent, 'notify');

        const idsToRemove = [];
        tasksListModel.taskListCollection.forEach(task => idsToRemove.push(task.id));
        tasksListModel.removeTask(idsToRemove);
        expect(tasksListModel.taskListCollection.length).toBeFalsy();
        expect(tasksListModel.settingsData.removeItem).toHaveBeenCalledWith(
            'tasks',
            idsToRemove
        );
        expect(tasksListModel.renderNoTasksLeftPageEvent.notify).toBeCalled();
    });

    it('should return falsy value', () => {
        const res = tasksListModel.removeTask();
        expect(res).toBeFalsy();
    });
});

describe('checkLeftTasks', () => {
    it('should call appropriate func', () => {
        jest.spyOn(tasksListModel.renderNoTasksLeftPageEvent, 'notify');

        tasksListModel.checkLeftTasks();
        expect(tasksListModel.renderNoTasksLeftPageEvent.notify).toBeCalled();
    });

    it('should call appropriate func', () => {
        jest.spyOn(tasksListModel.renderEmptyDailyTasks, 'notify');

        tasksListModel.taskListCollection.length = 1;
        tasksListModel.globalTasksActive.length = 1;
        tasksListModel.checkLeftTasks();
        expect(tasksListModel.renderEmptyDailyTasks.notify).toBeCalled();
    });

    it('should call appropriate func', () => {
        jest.spyOn(tasksListModel.renderAddFirstTaskPageEvent, 'notify');

        tasksListModel.taskListCollection.length = 1;
        tasksListModel.globalTasksActive.length = 0;
        tasksListModel.dailyTasksActive.length = 0;
        tasksListModel.dailyTasksCompleted.length = 0;
        tasksListModel.globalTasksCompleted.length = 0;

        tasksListModel.checkLeftTasks();
        expect(tasksListModel.renderAddFirstTaskPageEvent.notify).toBeCalled();
    });

    it('should call appropriate func', () => {
        jest.spyOn(tasksListModel.renderNoTasksLeftPageEvent, 'notify');

        tasksListModel.taskListCollection.length = 1;
        tasksListModel.globalTasksActive.length = 0;
        tasksListModel.dailyTasksActive.length = 0;
        tasksListModel.dailyTasksCompleted.length = 1;
        tasksListModel.globalTasksCompleted.length = 1;

        tasksListModel.checkLeftTasks();
        expect(tasksListModel.renderNoTasksLeftPageEvent.notify).toBeCalled();
    });
});

describe('getTaskById', () => {
    beforeAll(() => {
        tasksListModel.taskListCollection = tasks;
    });
    it('should return task by id', () => {
        const res = tasksListModel.getTaskById(tasks[0].id);
        expect(res).toEqual(tasks[0]);
    });

    it('should return falsy value', () => {
        const res = tasksListModel.getTaskById();
        expect(res).toBeFalsy();
    });
});

describe('saveTaskToStorage', () => {
    beforeAll(() => {
        tasksListModel.taskListCollection = tasks;
    });
    it('should call func to set value to storage', () => {
        jest.spyOn(tasksListModel.settingsData, 'setDataToStorage');
        tasksListModel.saveTaskToStorage(tasks[0].id);
        expect(tasksListModel.settingsData.setDataToStorage).toHaveBeenCalledWith(
            'active-task',
            tasks[0]
        );
    });

    it('should return falsy value', () => {
        const res = tasksListModel.saveTaskToStorage();
        expect(res).toBeFalsy();
    });
});
