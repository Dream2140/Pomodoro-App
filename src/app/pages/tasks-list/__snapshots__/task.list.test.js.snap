// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly allTasksDone 1`] = `"\\"<section class=\\\\\\"message message--exelent\\\\\\">\\\\r\\\\n    <div class=\\\\\\"message__container\\\\\\">\\\\r\\\\n        <div class=\\\\\\"message__text message__text--bottom-margin\\\\\\">\\\\r\\\\n            <p>Excellent,\\\\r\\\\n                all daily tasks done :)\\\\r\\\\n            </p>\\\\r\\\\n        </div>\\\\r\\\\n    </div>\\\\r\\\\n</section>\\""`;

exports[`rendering renders correctly emptyPage 1`] = `"\\"<section class=\\\\\\"message\\\\\\">\\\\r\\\\n    <div class=\\\\\\"message__container\\\\\\">\\\\r\\\\n        <img src=\\\\\\"../../images/tomato-addv02.svg\\\\\\" alt=\\\\\\"Logo\\\\\\" class=\\\\\\"message__img\\\\\\">\\\\r\\\\n        <div class=\\\\\\"message__text message__text--bottom-margin\\\\\\">\\\\r\\\\n            <p>Add your first task\\\\r\\\\n            </p>\\\\r\\\\n        </div>\\\\r\\\\n    </div>\\\\r\\\\n</section>\\""`;

exports[`rendering renders correctly firstPage 1`] = `"\\"<header class=\\\\\\"header \\\\\\">\\\\r\\\\n    <div class=\\\\\\"wrapper\\\\\\">\\\\r\\\\n        <div class=\\\\\\"header__container\\\\\\">\\\\r\\\\n\\\\r\\\\n            <div class=\\\\\\"header__logo logo\\\\\\">\\\\r\\\\n                <a href=\\\\\\"#\\\\\\" class=\\\\\\"logo__link\\\\\\">\\\\r\\\\n                    <img src=\\\\\\"/images/header.svg\\\\\\" alt=\\\\\\"header\\\\\\" class=\\\\\\"logo__img\\\\\\">\\\\r\\\\n                </a>\\\\r\\\\n            </div>\\\\r\\\\n\\\\r\\\\n            <div class=\\\\\\"header__title-wrapp\\\\\\">\\\\r\\\\n                <h1 class=\\\\\\"header__title\\\\\\">Daily task list</h1>\\\\r\\\\n            </div>\\\\r\\\\n\\\\r\\\\n            <nav class=\\\\\\"navigation header__nav\\\\\\">\\\\r\\\\n                <ul class=\\\\\\"navigation__list\\\\\\">\\\\r\\\\n                    <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                        <button class=\\\\\\"navigation__btn \\\\\\" data-route=\\\\\\"task-list\\\\\\" title=\\\\\\"Go to Tasks\\\\\\">\\\\r\\\\n                            <span class=\\\\\\"navigation__icon icon icon-tasks\\\\\\"></span>\\\\r\\\\n                        </button>\\\\r\\\\n                    </li>\\\\r\\\\n                    <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                        <button class=\\\\\\"navigation__btn \\\\\\" data-route=\\\\\\"reports/day/tasks\\\\\\" title=\\\\\\"Go to Reports\\\\\\">\\\\r\\\\n                            <span class=\\\\\\"navigation__icon icon icon-statistics\\\\\\"></span>\\\\r\\\\n                        </button>\\\\r\\\\n                    </li>\\\\r\\\\n                    <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                        <button class=\\\\\\"navigation__btn \\\\\\"\\\\r\\\\n                            data-route=\\\\\\"settings/pomodoros\\\\\\" title=\\\\\\"Go to Settings\\\\\\">\\\\r\\\\n                            <span class=\\\\\\"navigation__icon icon  icon-settings\\\\\\"></span>\\\\r\\\\n                        </button>\\\\r\\\\n                    </li>\\\\r\\\\n                </ul>\\\\r\\\\n            </nav>\\\\r\\\\n\\\\r\\\\n        </div>\\\\r\\\\n\\\\r\\\\n    </div>\\\\r\\\\n</header>    <main class=\\\\\\"content task-list-page\\\\\\">\\\\r\\\\n        <section class=\\\\\\"first-message message\\\\\\">\\\\r\\\\n            <div class=\\\\\\"wrapper\\\\\\">\\\\r\\\\n                <div class=\\\\\\"message__container\\\\\\">\\\\r\\\\n                    <img src=\\\\\\"../../images/tomato_settings.svg\\\\\\" alt=\\\\\\"Logo\\\\\\" class=\\\\\\"message__img\\\\\\">\\\\r\\\\n                    <div class=\\\\\\"message__text message__text--bottom-margin\\\\\\">\\\\r\\\\n                        <p>As you visited site for a first time you can\\\\r\\\\n                            check and customize your default\\\\r\\\\n                            application settings\\\\r\\\\n                        </p>\\\\r\\\\n                    </div>\\\\r\\\\n                    <div class=\\\\\\"message__buttons service-btn\\\\\\">\\\\r\\\\n                        <button class=\\\\\\"service-btn__btn service-btn__btn--blue message__buttons-skip\\\\\\">Skip</button>\\\\r\\\\n                        <button class=\\\\\\"service-btn__btn service-btn__btn--green message__buttons-settings\\\\\\">Go to settings</button>\\\\r\\\\n                    </div>\\\\r\\\\n                </div>\\\\r\\\\n            </div>\\\\r\\\\n        </section>\\\\r\\\\n\\\\r\\\\n\\\\r\\\\n    </main>\\\\r\\\\n\\""`;

exports[`rendering renders correctly noDailyTasks 1`] = `"\\"<section class=\\\\\\"message message--added-message\\\\\\">\\\\r\\\\n    <div class=\\\\\\"message__container\\\\\\">\\\\r\\\\n        <div class=\\\\\\"message__text\\\\\\">\\\\r\\\\n            <p>Task added,</p>\\\\r\\\\n            <p>Go it to the top 5 in daily task list</p>\\\\r\\\\n            <span class=\\\\\\"message__icon icon icon-arrow-top\\\\\\"></span>\\\\r\\\\n        </div>\\\\r\\\\n    </div>\\\\r\\\\n</section>\\""`;

exports[`rendering renders correctly noTasksLeft 1`] = `"\\"<section class=\\\\\\"message\\\\\\">\\\\r\\\\n    <div class=\\\\\\"message__container\\\\\\">\\\\r\\\\n        <img src=\\\\\\"../../images/tomato_settings.svg\\\\\\" alt=\\\\\\"Logo\\\\\\" class=\\\\\\"message__img\\\\\\">\\\\r\\\\n        <div class=\\\\\\"message__text message__text--bottom-margin\\\\\\">\\\\r\\\\n            <p>You donâ€™t have any tasks left. Add new tasks to stay productive.\\\\r\\\\n            </p>\\\\r\\\\n        </div>\\\\r\\\\n    </div>\\\\r\\\\n</section>\\""`;

exports[`rendering renders correctly taskListContainer 1`] = `"\\"<header class=\\\\\\"header \\\\\\">\\\\r\\\\n    <div class=\\\\\\"wrapper\\\\\\">\\\\r\\\\n        <div class=\\\\\\"header__container\\\\\\">\\\\r\\\\n\\\\r\\\\n            <div class=\\\\\\"header__logo logo\\\\\\">\\\\r\\\\n                <a href=\\\\\\"#\\\\\\" class=\\\\\\"logo__link\\\\\\">\\\\r\\\\n                    <img src=\\\\\\"/images/header.svg\\\\\\" alt=\\\\\\"header\\\\\\" class=\\\\\\"logo__img\\\\\\">\\\\r\\\\n                </a>\\\\r\\\\n            </div>\\\\r\\\\n\\\\r\\\\n            <div class=\\\\\\"header__title-wrapp\\\\\\">\\\\r\\\\n                <h1 class=\\\\\\"header__title\\\\\\">Daily task list</h1>\\\\r\\\\n            </div>\\\\r\\\\n\\\\r\\\\n            <nav class=\\\\\\"navigation header__nav\\\\\\">\\\\r\\\\n                <ul class=\\\\\\"navigation__list\\\\\\">\\\\r\\\\n                    <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                        <button class=\\\\\\"navigation__btn \\\\\\" data-route=\\\\\\"task-list\\\\\\" title=\\\\\\"Go to Tasks\\\\\\">\\\\r\\\\n                            <span class=\\\\\\"navigation__icon icon icon-tasks\\\\\\"></span>\\\\r\\\\n                        </button>\\\\r\\\\n                    </li>\\\\r\\\\n                    <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                        <button class=\\\\\\"navigation__btn \\\\\\" data-route=\\\\\\"reports/day/tasks\\\\\\" title=\\\\\\"Go to Reports\\\\\\">\\\\r\\\\n                            <span class=\\\\\\"navigation__icon icon icon-statistics\\\\\\"></span>\\\\r\\\\n                        </button>\\\\r\\\\n                    </li>\\\\r\\\\n                    <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                        <button class=\\\\\\"navigation__btn \\\\\\"\\\\r\\\\n                            data-route=\\\\\\"settings/pomodoros\\\\\\" title=\\\\\\"Go to Settings\\\\\\">\\\\r\\\\n                            <span class=\\\\\\"navigation__icon icon  icon-settings\\\\\\"></span>\\\\r\\\\n                        </button>\\\\r\\\\n                    </li>\\\\r\\\\n                </ul>\\\\r\\\\n            </nav>\\\\r\\\\n\\\\r\\\\n        </div>\\\\r\\\\n\\\\r\\\\n    </div>\\\\r\\\\n</header><main class=\\\\\\"content task-list-page\\\\\\">\\\\r\\\\n\\\\r\\\\n    <section class=\\\\\\"daily-content\\\\\\">\\\\r\\\\n        <div class=\\\\\\"wrapper\\\\\\">\\\\r\\\\n            <div class=\\\\\\"daily-content__container\\\\\\">\\\\r\\\\n\\\\r\\\\n                <div class=\\\\\\"tab \\\\\\">\\\\r\\\\n                    <div class=\\\\\\"tab__container\\\\\\">\\\\r\\\\n                        <ul class=\\\\\\"tab__item \\\\\\">\\\\r\\\\n                        </ul>\\\\r\\\\n                    </div>\\\\r\\\\n                </div>                <div class=\\\\\\"daily-content__content\\\\\\">\\\\r\\\\n\\\\r\\\\n                </div>\\\\r\\\\n                <div class=\\\\\\"daily-content__content--done d-none\\\\\\">\\\\r\\\\n\\\\r\\\\n                </div>\\\\r\\\\n\\\\r\\\\n            </div>\\\\r\\\\n        </div>\\\\r\\\\n    </section>\\\\r\\\\n    <section class=\\\\\\"global-list\\\\\\">\\\\r\\\\n        <div class=\\\\\\"wrapper\\\\\\">\\\\r\\\\n            <div class=\\\\\\"global-list__container\\\\\\">\\\\r\\\\n                <div class=\\\\\\"global-list__tabs tab\\\\\\">\\\\r\\\\n                    <div class=\\\\\\"tab__container\\\\\\">\\\\r\\\\n                        <div>\\\\r\\\\n                        <div class=\\\\\\"tab__item tab__item--global-btn\\\\\\">\\\\r\\\\n                            <button class=\\\\\\"tab__btn tab__btn--global\\\\\\" title=\\\\\\"Go to Global list\\\\\\">\\\\r\\\\n                                Global list\\\\r\\\\n                                <span class=\\\\\\"tab__arrow icon icon-arrow-right\\\\\\"></span>\\\\r\\\\n                            </button>\\\\r\\\\n                        </div>\\\\r\\\\n                        <ul class=\\\\\\"tab__item tab__item--left tab--hidden tab__item\\\\\\">\\\\r\\\\n                            <li class=\\\\\\"tab__btn tab__btn-select-global-list\\\\\\">Select all</li>\\\\r\\\\n                            <li class=\\\\\\"tab__btn tab__btn-disselect-global-list\\\\\\">Deselect all</li>\\\\r\\\\n                        </ul>\\\\r\\\\n                        </div>\\\\r\\\\n                        <ul class=\\\\\\"tab__item item--right tab__item--global\\\\\\">\\\\r\\\\n                            <li class=\\\\\\"tab__btn\\\\\\">All</li>\\\\r\\\\n                            <li class=\\\\\\"tab__btn\\\\\\">Urgent</li>\\\\r\\\\n                            <li class=\\\\\\"tab__btn\\\\\\">High</li>\\\\r\\\\n                            <li class=\\\\\\"tab__btn\\\\\\">Middle</li>\\\\r\\\\n                            <li class=\\\\\\"tab__btn\\\\\\">Low</li>\\\\r\\\\n                        </ul>\\\\r\\\\n                    </div>\\\\r\\\\n                </div>\\\\r\\\\n\\\\r\\\\n                <div class=\\\\\\"global-list__content\\\\\\">\\\\r\\\\n\\\\r\\\\n                </div>\\\\r\\\\n\\\\r\\\\n                <div class=\\\\\\"global-list__content--done d-none\\\\\\">\\\\r\\\\n\\\\r\\\\n                </div>\\\\r\\\\n            </div>\\\\r\\\\n        </div>\\\\r\\\\n    </section>\\\\r\\\\n\\\\r\\\\n\\\\r\\\\n</main>\\""`;
